dataland_analytics/analysis_scripts/scripts_with_api_admin/datarequests_by_filter.py

```diff
@@ -0,0 +1,156 @@
+            if all(req.get(key) == value for key, value in criterion.items()):
+                req["Request Status"] = "Pending"
+                break
+
+    return matches
+
+
+dataset_api_configuration = dataland_datasets.Configuration(
+    host=DATALAND_SERVER + "api", access_token=PERSONAL_USER_API_KEY
+)
+
+community_api_configuration = dataland_community.Configuration(
+    host=DATALAND_SERVER + "community", access_token=PERSONAL_USER_API_KEY
+)
+
+qa_api_configuration = dataland_qa.Configuration(
+    host=DATALAND_SERVER + "qa", access_token=PERSONAL_USER_API_KEY,
+)
+
+# Define filter parameters
+filter_parameters = {
+    "dataType": ["eutaxonomy-financials", "eutaxonomy-non-financials"],
+    "emailAddress": "piehl",
+}
+
+# Get requests by filter
+with dataland_community.ApiClient(community_api_configuration) as community_api_client:
+    request_controller_api_instance = dataland_community.RequestControllerApi(
+        api_client=community_api_client
+    )
+
+    requests_by_filter = request_controller_api_instance.get_data_requests(
+        data_type=filter_parameters["dataType"],
+        email_address=filter_parameters["emailAddress"],
+    )
+
+    requests_list = []
+
+    for request in requests_by_filter:
+        requests_list.append({
+            "Request ID": request.data_request_id,
+            "Company ID": request.dataland_company_id,
+            "Requester": request.user_email_address,
+            "Company": request.company_name,
+            "Framework": get_readable_string_from_datatype(request.data_type),
+            "Reporting Period": request.reporting_period,
+            "Date of Request": datetime.fromtimestamp(request.creation_timestamp/1000),
+            "Last Update of Request": datetime.fromtimestamp(request.last_modified_date/1000),
+            "Request Status": request.request_status.value,
+        })
+
+# Get pending dataset ids
+with dataland_qa.ApiClient(qa_api_configuration) as qa_api_client:
+    qa_controller_api_instance = dataland_qa.QaControllerApi(
+        api_client=qa_api_client
+    )
+
+    pending_dataset_ids = qa_controller_api_instance.get_unreviewed_datasets_ids()
+
+# Get metadata from pending dataset ids
+with dataland_datasets.ApiClient(dataset_api_configuration) as dataset_api_client:
+    non_financials_controller_api_instance = dataland_datasets.EutaxonomyNonFinancialsDataControllerApi(
+        api_client=dataset_api_client
+    )
+    metadata_controller_api_instance = dataland_datasets.MetaDataControllerApi(
+        api_client=dataset_api_client
+    )
+
+    pending_datasets_metadata = []
+    for dataset_id in pending_dataset_ids:
+        dataset = metadata_controller_api_instance.get_data_meta_info(dataset_id)
+        pending_datasets_metadata.append({
+            "Company ID": dataset.company_id,
+            "Reporting Period": dataset.reporting_period,
+            "Framework": get_readable_string_from_datatype(dataset.data_type),
```

Comment:
This function "get_readable_string_from_datatype" is in my opinion not necessary. I would just stick to the string of the Enum.